.github/workflows/cd.yml

```diff
@@ -151,64 +125,60 @@ jobs:
             --keychain $KEYCHAIN \
             $KEYCHAIN_PROFILE
 
-    - name: Prepare export for ${{ env.EXPORT_METHOD }}
-      if: ${{ matrix.destination.uploadto != 'ipa' }}
-      run: |
-        plutil -create xml1 ./export.plist
-        plutil -insert destination -string upload ./export.plist
-        plutil -insert method -string $EXPORT_METHOD ./export.plist
-
     - name: Prepare export for IPA
-      if: ${{ matrix.destination.uploadto == 'ipa' }}
+      if: matrix.platform == 'iOS' && env.UPLOAD_TO == 'ftp'
       run: |
         plutil -create xml1 ./export.plist
         plutil -insert method -string ad-hoc ./export.plist
         plutil -insert provisioningProfiles -dictionary ./export.plist
         plutil -replace provisioningProfiles -json '{ "self.Kiwix" : "iOS Team Provisioning Profile" }' ./export.plist
 
+    - name: Prepare export for ${{ matrix.uploadto }}
+      if: matrix.platform != 'iOS' || env.UPLOAD_TO == 'app-store'
```

Comment:
condition is hard to read ; can you add a comment describing the use cases?

Reply:
Please see it updated.